src/udev/udev-dump.c

```diff
@@ -28,64 +31,328 @@ void event_cache_written_sysctl(UdevEvent *event, const char *attr, const char *
         event_cache_written_value(&event->written_sysctls, attr, value);
 }
 
-void dump_event(UdevEvent *event, FILE *f) {
+static int dump_event_json(UdevEvent *event, sd_json_format_flags_t flags, FILE *f) {
         sd_device *dev = ASSERT_PTR(ASSERT_PTR(event)->dev);
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *v = NULL;
+        const char *str;
+        int r;
 
-        if (!f)
-                f = stdout;
+        if (sd_device_get_devpath(dev, &str) >= 0) {
+                r = sd_json_variant_set_field_string(&v, "path", str);
+                if (r < 0)
+                        return r;
+        }
 
-        if (!hashmap_isempty(event->written_sysattrs)) {
-                const char *key, *value;
+        if (sd_device_get_sysname(dev, &str) >= 0) {
+                r = sd_json_variant_set_field_string(&v, "name", str);
+                if (r < 0)
+                        return r;
+        }
 
-                fprintf(f, "%sWritten sysfs attributes:%s\n", ansi_highlight(), ansi_normal());
-                HASHMAP_FOREACH_KEY(value, key, event->written_sysattrs)
-                        fprintf(f, "  %s : %s\n", key, value);
+        unsigned sysnum;
+        if (sd_device_get_sysnum(dev, &str) >= 0 &&
+            safe_atou_full(str, SAFE_ATO_ALL_FLAGS, &sysnum) >= 0) {
```

Comment:
hmm, we have this same code at various places? maybe add a common internal helper for this?

`device_get_synum_as_unsigned()` or so?

the code exists at least in src/udev/udev-builtin-net_id.c, src/udev/udev-builtin-path_id.c

Reply:
OK. Added.